### PR TITLE
Implementation of isValid property

### PR DESCRIPTION
**Pull request by gsbelarus**
IsValid() function implemented for objects Client, Attachment, Transaction, Statement, ResultSet, BlobStream. So, we can test the object's validity after commit(), rollback(), dispose(), close() etc. methods have been invoked.

The newly added method is covered through the tests. In two places the tests have been adapted to be able to run on FB3.

Example of usage:
```ts
const transaction = await attachment.startTransaction();
try {
  ...
  // some code where the transaction could be committed or rolled back
  // but there are also conditions for the transaction to stay alive
  // until the code ends
  ...
} finally {
  if (transaction.isValid()) await transaction.commit();
}
```

### CODE REVIEW

File: packages/node-firebird-driver/src/test/tests.ts
Comment (asfernandes) [inline packages/node-firebird-driver/src/test/tests.ts:49]: Please take this thing out of this PR.

It's not completely correct in its usage and has nothing directly to do with the topic.

File: packages/node-firebird-driver/src/test/tests.ts
Comment (asfernandes) [inline packages/node-firebird-driver/src/test/tests.ts:307]: Please maintain PSQL code indented with spaces.

File: packages/node-firebird-driver/src/lib/index.ts
Comment (asfernandes) [inline packages/node-firebird-driver/src/lib/index.ts:1]: Please remove that interface. We don't want to have an inheritance thing in TS just to have a common method.

File: packages/node-firebird-driver-native/src/lib/blob.ts
Comment (asfernandes) [inline packages/node-firebird-driver-native/src/lib/blob.ts:91]: Please always put the return type in the method signature.

File: packages/node-firebird-driver/src/lib/impl/attachment.ts
Comment (asfernandes) [inline packages/node-firebird-driver/src/lib/impl/attachment.ts:202]: Please rewrite `check` methods to call `isValid` instead of duplicate the same conditions.

**Review by asfernandes — CHANGES_REQUESTED** [review]

**Comment by asfernandes** [comment]
Please add it to `Blob` and `Events`.

**Comment by asfernandes** [comment]
Please replace `isValid` to be a readonly property instead of a method. It looks more friendly.

File: packages/node-firebird-driver/src/test/tests.ts
Comment (gsbelarus) [reply packages/node-firebird-driver/src/test/tests.ts:49]: there are two places in tests which don't pass with FB3. 
one is INSERT INTO ... RETURNING *
and another NUMERIC(.., 20)

if you are ok with tests don't run under FB3, I will remove this check. I know that the better way is to get version of the server instead of ODS major version but how to obtain it?

Comment (asfernandes) [reply packages/node-firebird-driver/src/test/tests.ts:49]: But in the way you do, many valid tests for FB 3 will not run. So lets do it correctly in separate patch.

This code will also need adjustments for FB 5.

File: packages/node-firebird-driver/src/test/tests.ts
Comment (asfernandes) [inline packages/node-firebird-driver/src/test/tests.ts:296]: That changes is still weird.

I use following approach: indent with tabs up to the column where the command in string started. Then from there, indent with spaces. I also use VSCode and had no problem.

Please revert this.

**Comment by asfernandes** [comment]
Please set option "allow edits from maintainers" in this PR so I can send small edits before merge.

**Comment by gsbelarus** [comment]
It is set.

**Comment by asfernandes** [comment]
> It is set.

I still cannot push to it.

**Comment by asfernandes** [comment]
I have manually rebased your changes with mine and pushed to master, so closing this.

**Pull request closed without merging**
